Modules/Customer/Http/Services/CustomerService.php

```diff
@@ -100,6 +120,44 @@ public function update(array $requestData): array
             ];
         }
 
+        $nameArray = explode(' ', $requestData['cust_name']);
+        $first_name = $nameArray[0] ?? '';
+        $last_name = implode(' ', array_shift($nameArray));
+        $customer->first_name = $first_name;
+        $customer->last_name = $last_name;
+        $customer->shipping_name = $requestData['cust_name'];
+        $customer->email = $requestData['cust_email'];
+        $customer->store_name = $requestData['cust_storename'];
+        $customer->type = $requestData['cust_type'];
+        $customer->save();
+
+        return [
+            'res' => true,
+            'msg' => 'Your customer updated successfully',
+            'data' => $customer
+        ];
+    }
+
+    /**
+     * save the specified customer's details in storage.
+     *
+     * @param array $requestData
+     * @return array
+     */
+    public function save(array $requestData): array
+    {
+
+        $customer = Customer::where('customer_key', $requestData['customer_key'])->first();
+
+        // return error if no customer found
+        if (!$customer) {
+            return [
+                'res' => false,
+                'msg' => 'Customer not found !',
+                'data' => ""
+            ];
+        }
+
         $customer->update($requestData);
```

Comment:
@arijit1111 We are validating few of the parameters, but what if I pass a parameter other than the validated one? Lets say we are validating A,B,C but what if user passes D which haven't been validated through the Request while the column exists in Database, it'll directly update in the database and is harmful to security.